### ISSUE TITLE
Overriding CDARichHyperLink Doesn't propagate click events

### ISSUE THREAD

**Issue by GranthamAnthem** [contentful/rich-text-renderer-java#36]
Hey Folks,

I've overridden the styles for `CDARichHyperLink`, however, the click events no longer seem to be working. Hoping you all could help out with this?

```
// Override HyperLinks
        viewProcessor.overrideRenderer(
            {_, node ->
                node is CDARichHyperLink && node.content.size >= 1
            },
            { _, node ->
                val data = (node as CDARichHyperLink).data
                val content = (node.content.first() as CDARichText).text.toString()

                val myTextView = TextView(binding.staticContainer.context).apply {
                    layoutParams = ViewGroup.LayoutParams(
                        ViewGroup.LayoutParams.MATCH_PARENT,
                        ViewGroup.LayoutParams.WRAP_CONTENT
                    )
                    setText(spannable.invoke(content), TextView.BufferType.SPANNABLE)
                    movementMethod = LinkMovementMethod.getInstance()
                    setOnClickListener {
                        Timber.d("xxxxxxxxxx click something")
                    }
                }
                myTextView
            }
        )
```

**Comment by avneeshkumar1** [comment]
I am facing the same issue. did you find any solution?

**Comment by mthakuri** [comment]
In our requirement, we had to show the empty text/space added as part of the `RichText ` but removing `RemoveEmpties `was causing the HyperLink to not work. I have added below code that seems to be missing in the `BlockRenderer` causing this problem. I'm not sure if this answer solves the OP question but putting the finding here if someone else facing this issue.

![Issue](https://github.com/contentful/rich-text-renderer-java/assets/27849853/72cf5cf3-437d-44ea-9878-6abfc78d3ddb)

**Comment by GranthamAnthem** [comment]
@avneeshkumar1 I did something pretty hacky. Outside of the renderer when I've attached the output to a view, I made a function that recursively finds all instances of a TextView that has a link. I then return all potential Textviews and assign them a click listener. Works well but not ideal

**Comment by linavoda** [comment]
Hi, I am using the latest version 2.10 and still seeing this issue. Could this issued be reopened, please?

**Comment by linavoda** [comment]
@GranthamAnthem, have you found a solution for the problem you mentioned above? I am facing exactly the same issue with version 2.1.0.

**Comment by rafalniski** [comment]
@linavoda Can you please paste here your content model and the code you using?

**Comment by linavoda** [comment]
Hi, @rafalniski , thank you very much for your response.

This is the content model:
`
"{"contentBody":[{"data":{},"content":[{"data":{},"content":[{"data":{"uri":"http://google.com"},"content":[{"data":{},"marks":[],"value":"Check the website","nodeType":"text"}],"nodeType":"hyperlink"},{"data":{},"marks":[],"value":"","nodeType":"text"}],"nodeType":"paragraph"}],"nodeType":"document"}]}"
`

This is the code:
`    { _, node ->
                        val data = (node as CDARichHyperLink).data
                        val content = (node.content.first() as CDARichText).text.toString()

                        val aTextView = TextView(context).apply {
                            layoutParams = ViewGroup.LayoutParams(
                                    ViewGroup.LayoutParams.MATCH_PARENT,
                                    ViewGroup.LayoutParams.WRAP_CONTENT
                            )

                            val textContent = SpannableStringBuilder(content)

                            val clickableSpan: ClickableSpan = object : ClickableSpan() {
                                override fun onClick(textView: View) {
                                    Log.d("ContentHelper", "hello, span clicked")
                                }

                                override fun updateDrawState(ds: TextPaint) {
                                    ds.linkColor = Color.parseColor("#000000")
                                    super.updateDrawState(ds)
                                }
                            }
                            movementMethod = LinkMovementMethod.getInstance()
                            textContent.setSpan(clickableSpan, 0, textContent.length, 33)
                            text = textContent
                            movementMethod = LinkMovementMethod.getInstance()
                            setOnClickListener {
                                Log.d("ContentHelper", "hello, text view clicked")

                            }
                        }
                        aTextView
                    })`

**Comment by GranthamAnthem** [comment]
@linavoda yeah, so what I did is not ideal because the SDK should handle this, but it works! The renderer will iterate through each node of the CDARichDocument, and as it is passing through each node it will check if your condition is true before creating a native View of your choosing (in this case TextView). Once the render has finished, it creates and returns a LinearLayout of all processed nodes (at this point you will see your view rendered on the scree). 

From here, I created two helper functions that recursively search through a LinearLayout for all TextView's that contain the text of the link Im searching for. The first function does the recursive search for the text Im looking for (typically the content property). The second helper function just creates and returns a list of all TextViews. Now that I have a list of TextViews, I set a click listener on each TextView

I've quite possibly over-complicated this but given a tight deadline I couldn't find a quicker solution

**Comment by linavoda** [comment]
@GranthamAnthem, amazing, your solution works! Thank you very much again!

**Comment by linavoda** [comment]
@rafalniski , could we integrate @GranthamAnthem's fix as part of the library, so that we don't need to apply this fix in future versions, please?

**Comment by rafalniski** [comment]
@linavoda I released version `2.2.0` with the fixes. Could you please check if it works for you?

**Comment by linavoda** [comment]
@rafalniski, I have tested 2.2.0, it works! Thank you very much for fixing and releasing a new version so quickly.
